wordpress-mobile/WordPress-iOS
# Migrates GravatarServiceRemote classes & models to WordPressKit

Fixes #7310 

This PR:
* Migrates the `RemoteGravatarProfile ` model to the WPKit framework
* Migrates `GravatarServiceRemote` to the WPKit framework
* Migrates `NSMutableData+Helpers.swift` to the WPShared framework (used in main app and WPKit)
* Migrates `NSString+Helpers.h` & `NSString+Helpers.m` to the WPShared framework (used in main app and WPKit)
* Updates all imports for the files moved to the shared framework (and removed un-needed ones).

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. Verify you can log into an .org, non-jetpack site and the Gravitar is properly fetched [via](https://github.com/wordpress-mobile/WordPress-iOS/blob/eb7fdae0f6a3fb9dd093ceb88933fa137547fced/WordPress/Classes/ViewRelated/NUX/LoginSelfHostedViewController.swift#L319)
b. Verify you can upload a new Gravitar image via the Me tab

Needs review, por favor @elibud. Thanks!! 😄
